cmd/crictl/image.go

```diff
@@ -390,8 +390,11 @@ var removeImageCommand = &cli.Command{
 				return err
 			}
 			for _, img := range r {
-				logrus.Debugf("Adding image to be removed: %v", img.GetId())
-				ids[img.GetId()] = true
+				// Pinned images should not be removed on prune.
+				if all || (prune && !img.Pinned) {
+					logrus.Debugf("Adding image to be removed: %v", img.GetId())
+					ids[img.GetId()] = true
+				}
```

Comment:
What about something like this?

```suggestion
                                  // Pinned images should not be removed on prune.
                                  if prune && img.Pinned {
                                          logrus.Debugf("Excluding pinned container image: %v", img.GetId())
                                          continue
                                  }
                                  logrus.Debugf("Adding container image to be removed: %v", img.GetId())
                                  ids[img.GetId()] = true
```

I updated the wording to match other log messages.